src/read.js

```diff
@@ -4,13 +4,29 @@ var copychars = require("@dcos/copychars").default;
 
 var RECORD_PATTERN = /^\d+\n.+/;
 
+function countExtraCodepoints(string, start, end) {
+  let correctedLength = end - start;
```

Comment:
super minor, but what do you think of this


```
  let length = end - start;
  let correctedLength = length;
  for (let i = length; i >= start; i--) {
```

Reply:
Yes. I did not write it with legibility in mind, there is this and a few more I would like to fix if we are going to go for this solution. I also have to lint it and apply the same rigor we do with code on DC/OS.